subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/TaskNodeCodec.kt

```diff
@@ -101,10 +101,12 @@ class TaskNodeCodec(
                     writeReasonNotToTrackState(task)
                     beanStateWriterFor(task.javaClass).run {
                         writeStateOf(task)
-                        writeRegisteredPropertiesOf(
-                            task,
-                            this as BeanPropertyWriter
-                        )
+                        withTaskReferencesAllowed {
```

Comment:
@bamboo In fact we don't need to mutate isolate owner during reading

Reply:
Indeed 👍